File: problems/problems_2240/Solution.cpp
//go:build ignore
#include "cpp/common/Solution.h"

using namespace std;
using json = nlohmann::json;

class Solution {
public:
  long long waysToBuyPensPencils(int total, int cost1, int cost2) {
    int64_t ans = 0;
    while (total >= 0) {
      ans += total / cost2 + 1;
      total -= cost1;
    }
    return ans;
  }
};

json leetcode::qubh::Solve(string input_json_values) {
  vector<string> inputArray;
  size_t pos = input_json_values.find('\n');
  while (pos != string::npos) {
    inputArray.push_back(input_json_values.substr(0, pos));
    input_json_values = input_json_values.substr(pos + 1);
    pos = input_json_values.find('\n');
  }
  inputArray.push_back(input_json_values);

  Solution solution;
  int total = json::parse(inputArray.at(0));
  int cost1 = json::parse(inputArray.at(1));
  int cost2 = json::parse(inputArray.at(2));
  return solution.waysToBuyPensPencils(total, cost1, cost2);
}

File: problems/problems_2240/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.waysToBuyPensPencils(*test_input)

    def waysToBuyPensPencils(self, total: int, cost1: int, cost2: int) -> int:
        return sum((total - i * cost1) // cost2 + 1 for i in range(total // cost1 + 1))
